test/unit-test/test-Model.js

```diff
@@ -6493,6 +6493,123 @@ describe('Unit Test/Model Test', function() {
         });
       });
     });
+
+    it('"4-D tensor as input and output, the type of input and output being TENSOR_FLOAT32 type" are ok for "TANH" operation', function() {
+      return nn.createModel(options).then((model)=>{
+        let input = {type: nn.TENSOR_FLOAT32, dimensions: [1, 2, 2, 1]};
+        let output = {type: nn.TENSOR_FLOAT32, dimensions: [1, 2, 2, 1]};
+        model.addOperand(input);
+        model.addOperand(output);
+        assert.doesNotThrow(() => {
+          model.addOperation(nn.TANH, [0], [1]);
+        });
+      });
+    });
+
+    it('raise error when the rank of input and output are greater than 4 for "TANH" operation', function() {
+      return nn.createModel(options).then((model)=>{
+        let input = {type: nn.TENSOR_FLOAT32, dimensions: [1, 2, 2, 1, 2]};
+        let output = {type: nn.TENSOR_FLOAT32, dimensions: [1, 2, 2, 1, 2]};
+        model.addOperand(input);
+        model.addOperand(output);
+        assert.throws(() => {
+          model.addOperation(nn.TANH, [0], [1]);
+        });
+      });
+    });
+
+    it('raise error when the shape of input and output are different for "TANH" operation', function() {
+      return nn.createModel(options).then((model)=>{
+        let input = {type: nn.TENSOR_FLOAT32, dimensions: [1, 2, 2]};
+        let output = {type: nn.TENSOR_FLOAT32, dimensions: [1, 2]};
+        model.addOperand(input);
+        model.addOperand(output);
+        assert.throws(() => {
+          model.addOperation(nn.TANH, [0], [1]);
+        });
+      });
+    });
+
+    it('raise error when the type of input and output are FLOAT32 (not TENSOR_FLOAT32) for "TANH" operation', function() {
+      return nn.createModel(options).then((model)=>{
+        let input = {type: nn.FLOAT32};
+        let output = {type: nn.FLOAT32};
+        model.addOperand(input);
+        model.addOperand(output);
+        assert.throws(() => {
+          model.addOperation(nn.TANH, [0], [1]);
+        });
+      });
+    });
+
+    it('"4-D tensor as two inputs and one output, the type of input and output being TENSOR_FLOAT32 type" are ok for "MAXIMUM" operation', function() {
```

Comment:
@cuiyanx Please add more test case because MAXIMUM op does also support [TENSOR_INT32 and TENSOR_QUANT8_ASYMM type](
https://github.com/intel/webml-polyfill/blob/master/src/nn/Enums.js#L1235)